ci/ray_ci/automation/docker_tags_lib.py

```diff
@@ -0,0 +1,259 @@
+
+    url = "https://hub.docker.com/v2/users/login"
+    json_body = {
+        "username": username,
+        "password": password,
+    }
+    headers = {"Content-Type": "application/json"}
+    response = requests.post(url, headers=headers, json=json_body)
+    if response.status_code != 200:
+        raise AuthTokenException(f"Docker Hub. Error code: {response.status_code}")
+    return response.json().get("token", None)
+
+
+def _get_git_log(n_days: int = 30):
+    return subprocess.check_output(
+        [
+            "git",
+            "log",
+            f"--until='{n_days} days ago'",
+            "--pretty=format:%H",
+        ],
+        text=True,
+    )
+
+
+def list_recent_commit_short_shas(
+    n_days: int = 30,
+) -> List[str]:
+    """
+    Get list of recent commit SHAs (short version, first 6 char) on ray master branch.
+
+    Args:
+        n_days: Number of days to go back in git log.
+
+    Returns:
+        List of recent commit SHAs (6 char).
+    """
+    commit_shas = _get_git_log(n_days=n_days)
+    short_commit_shas = [
+        commit_sha[:6] for commit_sha in commit_shas.split("\n") if commit_sha
+    ]
+    return short_commit_shas
+
+
+def _call_crane_config(tag: str):
+    try:
+        return subprocess.check_output(
+            ["crane", "config", tag],
+            stderr=subprocess.STDOUT,
+            text=True,
+        )
+    except subprocess.CalledProcessError as e:
+        return e.output
+
+
+def get_image_creation_time(tag: str) -> datetime:
+    """
+    Get Docker image creation time from tag image config.
+
+    Args:
+        tag: Docker tag name
+
+    Returns:
+        Datetime object of image creation time.
+    """
+    result = _call_crane_config(tag=tag)
+    if "MANIFEST_UNKNOWN" in result or "created" not in result:
+        raise RetrieveImageConfigException()
+    config = json.loads(result)
+    return parser.isoparse(config["created"])
+
+
+def delete_tag(tag: str) -> bool:
+    """
+    Delete tag from Docker Hub repo.
+
+    Args:
+        tag: Docker tag name
+    Returns:
+        True if tag was deleted successfully, False otherwise.
+    """
+    token = get_docker_hub_auth_token()
+    headers = {
+        "Authorization": f"Bearer {token}",
+    }
+    namespace, repo_tag = tag.split("/")
+    repository, tag_name = repo_tag.split(":")
+
+    url = f"https://hub.docker.com/v2/repositories/{namespace}/{repository}/tags/{tag_name}"  # noqa E501
+    response = requests.delete(url, headers=headers)
+    if response.status_code == 429:
+        raise DockerHubRateLimitException()
+    if response.status_code != 204:
+        logger.info(f"Failed to delete {tag}, status code: {response.json()}")
+        return False
+    else:
+        logger.info(f"Deleted tag {tag}")
+        return True
+
+
+def query_tags_from_docker_hub(
+    filter_func: Callable[[str], bool],
+    namespace: str,
+    repository: str,
+    num_tags: Optional[int] = 1000,
```

Comment:
set this to None for default. and if you want to set another default, you can `if num_tags is None: num_tags = 1000`

`num_tags = 0` or `-1` for no max, if there is a such a use case.

Reply:
changed to `None`. For no max, I use an if statement `if num_tags` which should cover case where `num_tags=None` and `num_tags=0` (which means no max) don't apply.